crates/abq_queue/src/persistence/manifest/fs.rs

```diff
@@ -42,7 +54,14 @@ impl PersistentManifest for FilesystemPersistor {
 
         let packed = serde_json::to_vec(&view).located(here!())?;
 
-        fs::write(path, packed).await.located(here!())?;
+        fs::write(&path, packed).await.located(here!())?;
+
+        // We must write to the remote only after the content has been written to disk, since the
+        // upload to remote will read from disk.
+        // TODO(PERF): write to remote in parallel by passing the in-memory bytes instead.
+        self.remote
+            .store(PersistenceKind::Manifest, run_id, &path)
+            .await?;
```

Comment:
What do you think we should do if this fails? It'd be nice to retry it a few times with backoff in the background (does the S3 remote impl already do that?)

Reply:
S3 will retry twice. Yeah, we should probably retry with backoff.